### PR TITLE
FROST key repair

### PR DESCRIPTION
**Pull request by wydengyre**
A rudimentary implementation of FROST key repair, based on the ZCash FROST repair implementation.

### CODE REVIEW

File: protocols/frost/repair/round1.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+// We need Lj(x) where x is the id of the lost party.
+//
+// The Lagrange polynomial for a set of points (x_j, y_j) for 0 <= j <= k
+// is ∑_{i=0}^k y_i.ℓ_i(x), where ℓ_i(x) is the Lagrange basis polynomial:
+//
+// ℓ_i(x) = ∏_{0≤j≤k; j≠i} (x - x_j) / (x_i - x_j).
+//
+// This computes ℓ_j(x) for the set of points `xs` and for the j corresponding
+// to the given xj.
+func computeLagrangeCoefficient(group curve.Curve, helpers party.IDSlice, lostShare party.ID, selfID party.ID) curve.Scalar {
```
Comment (blangel) [inline protocols/frost/repair/round1.go:124]: @wydengyre is this not done already in the `multi-party-sig` library for some other FROST function? I.e. I'm surprised this has to be ported from the ZCash library.

Comment (wydengyre) [reply protocols/frost/repair/round1.go:124]: The function's first comment explains this:

```
// The lagrange functions in the polynomial package only compute Lj(0).
// We need Lj(x) where x is the id of the lost party.
```

Comment (blangel) [reply protocols/frost/repair/round1.go:124]: Ok, I think I read that sentence without understanding. I think `...only computes` would clarify slightly (but no need to change simply for that)

Comment (blangel) [reply protocols/frost/repair/round1.go:124]: I see. I had a hard time reading that sentence twice. First thinking it was not plural and should be `computes` but then realizing it was plural.

File: protocols/frost/repair/round3.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		// we return zero scalar for consistency
+		return r.ResultRound(r.Group().NewScalar()), nil
+	}
+
+	share := r.Group().NewScalar()
+	for _, sigma := range r.sigmas {
+		share = share.Add(sigma)
+	}
+
+	// TODO: deal with public commitment share
```
Comment (blangel) [inline protocols/frost/repair/round3.go]: @wydengyre is this a TODO from the Zcash or something we need to address?

Comment (wydengyre) [reply protocols/frost/repair/round3.go]: I should get rid of this TODO. It's not something we need to address.

Comment (wydengyre) [reply protocols/frost/repair/round3.go]: force pushed an amended commit that kills the TODO

**Review by blangel — APPROVED** [review]

**Pull request merged**